numba_dpex/core/parfors/kernel_parfor_pass.py

```diff
@@ -0,0 +1,103 @@
+from typing import NamedTuple, Union
```

Comment:
license header is missing

Reply:
I also think `parfor_kernel_pass` or `parfor_to_kernel_pass` is a better name as a parfor is being converted to a kernel.

Reply:
I guess `parfor_inject_kernel_pass` is a better name